source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -180,15 +180,22 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers) {
   // Now that we'll definitely be making the request, add filter state stats if configured to do so.
   const Envoy::StreamInfo::FilterStateSharedPtr& filter_state =
       decoder_callbacks_->streamInfo().filterState();
-  if ((config_->emitFilterStateStats() || config_->filterMetadata().has_value()) &&
-      !filter_state->hasData<ExtAuthzLoggingInfo>(decoder_callbacks_->filterConfigName())) {
-    filter_state->setData(decoder_callbacks_->filterConfigName(),
-                          std::make_shared<ExtAuthzLoggingInfo>(config_->filterMetadata()),
-                          Envoy::StreamInfo::FilterState::StateType::Mutable,
-                          Envoy::StreamInfo::FilterState::LifeSpan::Request);
-
-    logging_info_ =
-        filter_state->getDataMutable<ExtAuthzLoggingInfo>(decoder_callbacks_->filterConfigName());
+  if ((config_->emitFilterStateStats() || config_->filterMetadata().has_value())) {
+    if (!filter_state->hasDataWithName(decoder_callbacks_->filterConfigName())) {
```

Comment:
Could you please elaborate why you change from `hasData` to `hasDataWithName`?

hasData won't be able to detect the name collision?

Reply:
There could be data of a different type under this name.

Reply:
Hmm, does this mean that if you keep the code as it is `hasData<ExtAuthzLoggingInfo>`, there will be no name collision, as the data under same name is a different data type?

Reply:
I'm not sure if the same name can contain multiple types, checking...

Reply:
https://github.com/envoyproxy/envoy/blob/main/source/common/stream_info/filter_state_impl.h#L57
No, the map is only from name to object.

Reply:
We can either override preexisting filter state data or we can preserve it and not emit the ExtAuthzLoggingInfo. I chose to preserve preexisting data. If that's the desired behavior, I don't think it matters what type it is.

Reply:
IIUC, we are technically "creating" nullptr access here? because of switching from `hasData ` to `hasDataWithName ` to preserve pre-existing data.

I actually would think the opposite direction: we should honor the ExtAuthzLoggingInfo here as it is created by the filter with which config name is associated.  Or in which case in reality you think other filter should pre-occupy filter state with ext_authz's filter config name.